cmd/cli/version_test.go

```diff
@@ -0,0 +1,188 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/pkg/errors"
+	tassert "github.com/stretchr/testify/assert"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/version"
+)
+
+type fakeRemoteVersion struct {
+	err         error
+	versionInfo *version.Info
+}
+
+func (f *fakeRemoteVersion) proxyGetMeshVersion(pod string, namespace string, clientset kubernetes.Interface) (*version.Info, error) {
+	if f.err != nil {
+		return nil, f.err
+	}
+	return f.versionInfo, nil
+}
+
+func TestGetMeshVersion(t *testing.T) {
+	tests := []struct {
+		name              string
+		namespace         string
+		remoteVersion     *version.Info
+		remoteVersionInfo *remoteVersionInfo
+		controllerPods    []*corev1.Pod
+		err               error
+	}{
+		{
+			name:              "no mesh in namespace",
+			namespace:         "test",
+			remoteVersion:     nil,
+			remoteVersionInfo: nil,
+			controllerPods:    []*corev1.Pod{},
+			err:               errors.Errorf("No mesh found in namespace [%s]", "test"),
+		},
+		{
+			name:              "mesh in namespace and proxyGetMeshVersion fails",
+			namespace:         "test",
+			remoteVersion:     nil,
+			remoteVersionInfo: nil,
+			controllerPods: []*corev1.Pod{
+				{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "controllerPod",
+						Namespace: "test",
+						Labels: map[string]string{
+							constants.AppLabel:               constants.OSMControllerName,
+							constants.OSMAppInstanceLabelKey: "osm",
+						},
+					},
+				},
+			},
+			err: errors.Errorf("Error retrieving mesh version from pod %s in namespace %s)", "controllerPod", settings.Namespace()),
```

Comment:
I find inlining string constants in format strings easier to read than making them format parameters.
```suggestion
			err: errors.Errorf("Error retrieving mesh version from pod controllerPod in namespace %s", settings.Namespace()),
```